pkg/docker/docker.go

```diff
@@ -552,7 +490,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*api.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && !strings.Contains(err.(errors.Error).Details.Error(), "no such image") {
+	if err != nil && !strings.Contains(err.(errors.Error).Details.Error(), "No such image") {
```

Comment:
Yes, that's right.  I wanted to write !dockerapi.IsErrImageNotFound here, which works except for in the test harness: it's not possible to synthesise an imageNotFoundError for testing because it's not public within the engine API...